### PR TITLE
Assign port, http schema automatically based on passed hostname

### PR DESCRIPTION
**Pull request by mdesmet**
## Release notes

( ) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
(*) Release notes are required, with the following suggested text:

```markdown
* Assign port, http schema automatically based on passed hostname
```

### CODE REVIEW

File: tests/unit/test_dbapi.py [hunk 1]
```diff
@@ -272,3 +272,35 @@ def test_role_is_set_when_specified(mock_client):
...
+    assert http_server_without_port.host == "mytrinoserver.domain"
+    assert http_server_without_port.port == 8080
+    assert http_server_without_port.http_scheme == constants.HTTP
+
+    http_server_with_path = Connection("http://mytrinoserver.domain/some_path")
+    assert http_server_with_path.host == "mytrinoserver.domain/some_path"
+    assert http_server_with_path.port == 8080
+    assert http_server_with_path.http_scheme == constants.HTTP
+
+    only_hostname = Connection("mytrinoserver.domain")
```
Comment (hashhar) [inline tests/unit/test_dbapi.py:303]: Does `only_hostname_with_path` also work?

**Review by hashhar — APPROVED** [review]
LGTM % comment

File: tests/unit/test_dbapi.py (hunk 1 above)
Comment (mdesmet) [reply tests/unit/test_dbapi.py:303]: Added additional test case.

**Comment by leniartek** [comment]
@hashhar @findepi can one of you merge this please?
This is blocker for dbt-trino release to dbt Cloud.

**Pull request merged**